docs/multicast-guide.md

```diff
@@ -0,0 +1,122 @@
+# Multicast User Guide
+
+Antrea supports Multicast traffic in the following scenarios:
+
+1. Pods to Pods, any Pod that has joined the multicast group should receive the traffic from the Pod senders send multicast traffic to that group.
+2. Pods to External, external hosts connected to the Node will receive the multicast traffic from that local Pods.
+3. External to Pod, multicast traffic sent by external hosts connected to the Nodes will be received if some local Pods have joined the multicast group.
```

Comment:
Use active voice to be consistent with the first two points: "Pods can receive the multicast traffic from external hosts."

Reply:
I feel no need to repeat "joined the multicast group" every time. Readers understand that.

Reply:
updated